<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>plant_model</class>
 <widget class="QDialog" name="plant_model">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1063</width>
    <height>879</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Plant Model Parameters Configuration</string>
  </property>
  <widget class="QPushButton" name="save_button">
   <property name="geometry">
    <rect>
     <x>770</x>
     <y>850</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </widget>
  <widget class="QPushButton" name="close_button">
   <property name="geometry">
    <rect>
     <x>970</x>
     <y>850</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Close</string>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>60</y>
     <width>371</width>
     <height>211</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;img src=&quot;:/model_config/images/dc_motor.png&quot;/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>70</y>
     <width>371</width>
     <height>191</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;img src=&quot;:/model_config/images/dc_transfer_function.png&quot;/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>10</y>
     <width>711</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Times New Roman</family>
     <pointsize>11</pointsize>
     <weight>75</weight>
     <italic>true</italic>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>This block is used to configure the plant model.
The default is a DC motor, or you can check the box â€œcustom modeâ€ below to use 
 your own transfer function.
</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="dc_motor_box">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>270</y>
     <width>721</width>
     <height>331</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>9</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="title">
    <string>DC motor parameters</string>
   </property>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>701</width>
      <height>301</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0" alignment="Qt::AlignLeft">
      <widget class="QLabel" name="label_4">
       <property name="font">
        <font>
         <family>Times New Roman</family>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Armature Inductance (L)</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_11">
       <property name="font">
        <font>
         <family>Times New Roman</family>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Load torque (Td)</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1" alignment="Qt::AlignHCenter">
      <widget class="QLineEdit" name="motor_L"/>
     </item>
     <item row="1" column="1" alignment="Qt::AlignHCenter">
      <widget class="QLineEdit" name="motor_R"/>
     </item>
     <item row="6" column="1" alignment="Qt::AlignHCenter">
      <widget class="QLineEdit" name="motor_Kb"/>
     </item>
     <item row="2" column="1" alignment="Qt::AlignHCenter">
      <widget class="QLineEdit" name="motor_Km"/>
     </item>
     <item row="5" column="1" alignment="Qt::AlignHCenter">
      <widget class="QLineEdit" name="motor_Kf"/>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_14">
       <property name="font">
        <font>
         <family>Times New Roman</family>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Back-EMF Constant (Kb)</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_12">
       <property name="font">
        <font>
         <family>Times New Roman</family>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Rotor and load inertia (J)</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1" alignment="Qt::AlignHCenter">
      <widget class="QLineEdit" name="motor_J"/>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_13">
       <property name="font">
        <font>
         <family>Times New Roman</family>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Viscous friction coefficient(Kf)
</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_5">
       <property name="font">
        <font>
         <family>Times New Roman</family>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Armature Resistance (R)</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1" alignment="Qt::AlignHCenter">
      <widget class="QLineEdit" name="motor_Td"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_10">
       <property name="font">
        <font>
         <family>Times New Roman</family>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Torque Constant (Km)</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLabel" name="label_15">
       <property name="font">
        <font>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>[Î©]</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLabel" name="label_16">
       <property name="font">
        <font>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>[H]</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QLabel" name="label_17">
       <property name="font">
        <font>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>[N.m/A]</string>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QLabel" name="label_18">
       <property name="font">
        <font>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>[N.m]</string>
       </property>
      </widget>
     </item>
     <item row="4" column="2">
      <widget class="QLabel" name="label_19">
       <property name="font">
        <font>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>[kg.mÂ²]</string>
       </property>
      </widget>
     </item>
     <item row="5" column="2">
      <widget class="QLabel" name="label_20">
       <property name="font">
        <font>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>[N.m.s/rad]</string>
       </property>
      </widget>
     </item>
     <item row="6" column="2">
      <widget class="QLabel" name="label_21">
       <property name="font">
        <font>
         <pointsize>9</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>[VÂ·s/rad]</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QCheckBox" name="default_box">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>260</y>
     <width>171</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Times New Roman</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Default Parameters</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="custom_box">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>600</y>
     <width>171</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Times New Roman</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Custom mode</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="custom_mode_box">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>620</y>
     <width>711</width>
     <height>221</height>
    </rect>
   </property>
   <property name="title">
    <string/>
   </property>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>-1</x>
      <y>9</y>
      <width>711</width>
      <height>243</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_22">
       <property name="font">
        <font>
         <family>Times New Roman</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Enter the numerator and denominator coefficients
in the format:
[a1, a2, a3,â€¦ an] 
Where F(s) = ğ‘_1 ğ‘ ^(ğ‘›âˆ’1)+ğ‘_2 ğ‘ ^(ğ‘›âˆ’2)+ â€¦+ğ‘_ğ‘›
For example:
G(s) = (ğ‘›ğ‘¢ğ‘š(ğ‘ ))/(ğ‘‘ğ‘’ğ‘›(ğ‘ )) 
= (5ğ‘ +10)/(12ğ‘ ^2+3ğ‘ +7)
num = [5, 10]
den = [12, 3, 7]
</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="1" column="0" alignment="Qt::AlignRight">
        <widget class="QLabel" name="label_23">
         <property name="font">
          <font>
           <family>Times New Roman</family>
           <pointsize>9</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Den coefficients</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" alignment="Qt::AlignRight">
        <widget class="QLabel" name="label_24">
         <property name="font">
          <font>
           <family>Times New Roman</family>
           <pointsize>9</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Num coefficients</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1" alignment="Qt::AlignLeft">
        <widget class="QLineEdit" name="den_custom_coeff"/>
       </item>
       <item row="0" column="1" alignment="Qt::AlignLeft">
        <widget class="QLineEdit" name="num_custom_coeff"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="display_result">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>280</y>
     <width>321</width>
     <height>321</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Times New Roman</family>
     <pointsize>9</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="title">
    <string/>
   </property>
   <widget class="QLabel" name="transfer_function_res">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>90</y>
      <width>311</width>
      <height>241</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;img src=&quot;:/model_config/images/tf_mockup.png&quot;/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_25">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>321</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>13</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Transfer function based &lt;/p&gt;&lt;p align=&quot;center&quot;&gt;on the entered parameters&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
  </widget>
  <widget class="QPushButton" name="apply_button">
   <property name="geometry">
    <rect>
     <x>870</x>
     <y>850</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Apply</string>
   </property>
  </widget>
 </widget>
 <resources>
  <include location="../resources/plant_model.qrc"/>
 </resources>
 <connections/>
</ui>
